### PR TITLE
[Core/Achievement] Fix Save The Day

### PR DESCRIPTION
**Pull request by Laintime**
Closes https://github.com/TrinityCore/TrinityCore/issues/15202

### CODE REVIEW

**Comment by Rushor** [comment]
https://ariejan.net/2011/07/05/git-squash-your-latests-commits-into-one/ if you want to update it @Laintime :D

File: src/server/game/Battlegrounds/Zones/BattlegroundWS.cpp [hunk 1]
```diff
@@ -882,7 +882,10 @@ bool BattlegroundWS::CheckAchievementCriteriaMeet(uint32 criteriaId, Player cons
     switch (criteriaId)
     {
         case BG_CRITERIA_CHECK_SAVE_THE_DAY:
-            return GetFlagState(player->GetTeam()) == BG_WS_FLAG_STATE_ON_BASE;
+            if(target && target->GetTypeId() == TYPEID_PLAYER)
```
Comment (Vincent-Michael) [inline src/server/game/Battlegrounds/Zones/BattlegroundWS.cpp]: i like more:  "if (Player* playerTarget = target->ToPlayer())"

File: src/server/game/Battlegrounds/Zones/BattlegroundWS.cpp [hunk 2]
```diff
@@ -882,7 +882,10 @@ bool BattlegroundWS::CheckAchievementCriteriaMeet(uint32 criteriaId, Player cons
     switch (criteriaId)
     {
         case BG_CRITERIA_CHECK_SAVE_THE_DAY:
-            return GetFlagState(player->GetTeam()) == BG_WS_FLAG_STATE_ON_BASE;
+            if(target && target->GetTypeId() == TYPEID_PLAYER) 
+           {
+                return GetFlagState(target->ToPlayer()->GetTeam()) == BG_WS_FLAG_STATE_ON_BASE;
+            }
```
Comment (tkrokli) [inline src/server/game/Battlegrounds/Zones/BattlegroundWS.cpp]: ... and as Carbenium wrote, just remove the bracket lines. Keep the indentation on the `return GetFlagState` line as it is.

Comment (joschiwald) [inline src/server/game/Battlegrounds/Zones/BattlegroundWS.cpp]: return false missing here

File: src/server/game/Battlegrounds/Zones/BattlegroundWS.cpp (hunk 1 above)
Comment (sirikfoll) [inline src/server/game/Battlegrounds/Zones/BattlegroundWS.cpp]: this line is really necessary? can't think on a situation where this check will fail

Comment (Keader) [inline src/server/game/Battlegrounds/Zones/BattlegroundWS.cpp]: In this case, if target is null, would not crashing? ("&&" Do not verify all conditions?)

Comment (Runico) [inline src/server/game/Battlegrounds/Zones/BattlegroundWS.cpp]: @Keader, In this case, if playerTarget is null the next line will never execute

**Comment by jackpoz** [comment]
please check the comments and update your PR

**Pull request closed without merging**